### PR TITLE
Clean up helix retry list

### PR DESCRIPTION
**Pull request by amcasey**
I'd like to figure out what's actually failing for each flaky test so I can file an appropriately bug.  Anything determined to be inherently flaky can be re-added.

### CODE REVIEW

File: src/Hosting/test/FunctionalTests/WebHostBuilderTests.cs [hunk 1]
```diff
@@ -14,6 +14,7 @@ public class WebHostBuilderTests : LoggedTest
     public static TestMatrix TestVariants => TestMatrix.ForServers(ServerType.Kestrel)
             .WithTfms(Tfm.Default);
 
+    [Repeat(100)]
```
Comment (amcasey) [inline src/Hosting/test/FunctionalTests/WebHostBuilderTests.cs:17]:
```suggestion
    [QuarantineTest("https://github.com/dotnet/aspnetcore/issues/52429")]
```

File: src/Components/Components/test/NavigationManagerTest.cs [hunk 2]
```diff
@@ -797,6 +797,7 @@ async ValueTask HandleLocationChanging_ThrowException(LocationChangingContext co
         }
     }
 
+    [QuarantineTest("https://github.com/dotnet/aspnetcore/issues/52407")]
```
Comment (amcasey) [inline src/Components/Components/test/NavigationManagerTest.cs:800]:
```suggestion
    [QuarantinedTest("https://github.com/dotnet/aspnetcore/issues/52407")]
```

File: src/SignalR/clients/csharp/Client/test/FunctionalTests/HubConnectionTests.cs [hunk 3]
```diff
@@ -2667,6 +2667,7 @@ await using (var server = await StartServer<Startup>(w => w.EventId.Name == "Rec
         }
     }
 
+    [QuarantineTest("https://github.com/dotnet/aspnetcore/issues/52408")]
```
Comment (amcasey) [inline src/SignalR/clients/csharp/Client/test/FunctionalTests/HubConnectionTests.cs:2670]:
```suggestion
    [QuarantinedTest("https://github.com/dotnet/aspnetcore/issues/52408")]
```

File: src/Servers/Kestrel/Transport.NamedPipes/test/NamedPipeConnectionTests.cs [hunk 4]
```diff
@@ -14,6 +14,7 @@ public class NamedPipeConnectionTests : TestApplicationErrorLoggerLoggedTest
 {
     private static readonly byte[] TestData = Encoding.UTF8.GetBytes("Hello world");
 
+    [Repeat(100)]
```
Comment (amcasey) [inline src/Servers/Kestrel/Transport.NamedPipes/test/NamedPipeConnectionTests.cs:17]:
```suggestion
    [QuarantinedTest("https://github.com/dotnet/aspnetcore/issues/52462")]
```

File: src/Servers/Kestrel/test/Interop.FunctionalTests/HttpClientHttp2InteropTests.cs [hunk 5]
```diff
@@ -689,6 +689,7 @@ public async Task ServerReset_AfterTrailers_NoError(string scheme)
         await host.StopAsync().DefaultTimeout();
     }
 
+    [Repeat(100)]
```
Comment (amcasey) [inline src/Servers/Kestrel/test/Interop.FunctionalTests/HttpClientHttp2InteropTests.cs:692]:
```suggestion
    [QuarantinedTest("https://github.com/dotnet/aspnetcore/issues/46717")]
```

File: src/Servers/Kestrel/test/FunctionalTests/ResponseTests.cs [hunk 6]
```diff
@@ -417,14 +417,15 @@ public async Task AppCanHandleClientAbortingConnectionMidResponse(ListenOptions
         Assert.Empty(transportLogs.Where(w => w.LogLevel > LogLevel.Debug));
     }
 
+    [Repeat(100)]
```
Comment (amcasey) [inline src/Servers/Kestrel/test/FunctionalTests/ResponseTests.cs:420]:
```suggestion
    [QuarantinedTest("https://github.com/dotnet/aspnetcore/issues/52464")]
```

**Comment by amcasey** [comment]
Final tally:

Quarantined
- LocationChangingHandlers_CannotCancelTheNavigationAsynchronously_UntilReturning
- InjectedStartup_DefaultApplicationNameIsEntryAssembly
- BidirectionalStream_ServerReadsDataAndCompletes_GracefullyClosed (named pipes)
- ClientAbortingConnectionImmediatelyIsNotLoggedHigherThanDebug
- ServerReset_BeforeRequestBody_ClientBodyThrows
- ChangingUserNameDuringReconnectLogsWarning

Un-retried
- CanLaunchPhotinoWebViewAndClickButton
- WebConfigAppendsHostingStartup_OutOfProcess
- StreamPool_StreamAbortedOnClientAndServer_NotPooled
- BidirectionalStream_ServerReadsDataAndCompletes_GracefullyClosed (quic)
- HttpsConnectionClosedWhenResponseDoesNotSatisfyMinimumDataRate
- TraceIdentifierIsUnique

File: src/Components/WebView/test/E2ETest/WebViewManagerE2ETests.cs [hunk 7]
```diff
@@ -15,6 +15,7 @@ public class WebViewManagerE2ETests
     //   There's probably some way to make it work, but it's not currently a supported Blazor Hybrid scenario anyway
     // - macOS is skipped due to the test not being able to detect when the WebView is ready. There's probably an issue
     //   with the JS code sending a WebMessage to C# and not being sent properly or detected properly.
+    [Repeat(100)]
```
Comment (amcasey) [inline src/Components/WebView/test/E2ETest/WebViewManagerE2ETests.cs:18]:
```suggestion
    [QuarantinedTest("https://github.com/dotnet/aspnetcore/issues/50802")]
```

File: src/Servers/Kestrel/test/FunctionalTests/ResponseTests.cs [hunk 8]
```diff
@@ -585,6 +586,7 @@ async Task App(HttpContext context)
         }
     }
 
+    [Repeat(100)]
```
Comment (amcasey) [inline src/Servers/Kestrel/test/FunctionalTests/ResponseTests.cs:589]:
```suggestion
```

File: src/Servers/IIS/IIS/test/Common.LongTests/StartupTests.cs [hunk 9]
```diff
@@ -1220,6 +1220,7 @@ private async Task WebConfigOverridesGlobalEnvironmentVariables(HostingModel hos
     [RequiresIIS(IISCapability.PoolEnvironmentVariables)]
     public Task WebConfigAppendsHostingStartup_InProcess() => WebConfigAppendsHostingStartup(HostingModel.InProcess);
 
+    [Repeat(100)]
```
Comment (amcasey) [inline src/Servers/IIS/IIS/test/Common.LongTests/StartupTests.cs:1223]:
```suggestion
```

File: src/Servers/Kestrel/Transport.Quic/test/QuicConnectionContextTests.cs [hunk 10]
```diff
@@ -441,6 +441,7 @@ public async Task StreamPool_StreamAbortedOnClient_NotPooled()
         Assert.Equal(0, quicConnectionContext.StreamPool.Count);
     }
 
+    [Repeat(100)]
```
Comment (amcasey) [inline src/Servers/Kestrel/Transport.Quic/test/QuicConnectionContextTests.cs:444]:
```suggestion
```

File: src/Servers/Kestrel/Transport.Quic/test/QuicStreamContextTests.cs [hunk 11]
```diff
@@ -25,6 +25,7 @@ public class QuicStreamContextTests : TestApplicationErrorLoggerLoggedTest
 {
     private static readonly byte[] TestData = Encoding.UTF8.GetBytes("Hello world");
 
+    [Repeat(100)]
```
Comment (amcasey) [inline src/Servers/Kestrel/Transport.Quic/test/QuicStreamContextTests.cs:28]:
```suggestion
```

File: src/Servers/Kestrel/test/InMemory.FunctionalTests/RequestTests.cs [hunk 12]
```diff
@@ -582,6 +582,7 @@ public async Task AppCanSetTraceIdentifier()
         }
     }
 
+    [Repeat(100)]
```
Comment (amcasey) [inline src/Servers/Kestrel/test/InMemory.FunctionalTests/RequestTests.cs:585]:
```suggestion
```

**Review by amcasey — COMMENTED** [review]
Remove RepeatAttribute

**Review by BrennanConroy — APPROVED** [review]

**Comment by JamesNK** [comment]
@amcasey Love it.

IMO the retry list should be a solution of last resort for tests that are inherently flaky. Should purge it as much as possible.

Does the JSON file support comments? It would be great to have a comment about why something is in this list. e.g.

```json
{
  "version" : 1,
  "defaultOnFailure": "fail",
  "localRerunCount" : 3,
  "retryOnRules": [
    {"testName": {"contains": "FlakyTestToEnsureRetryWorks" }}, // Sanity check
    {"testName": {"contains": "ConfigureEndpointDevelopmentCertificateGetsLoadedWhenPresent" }}, // Retry required because ABC
    {"testName": {"contains": "MaxRequestBufferSizeTests.LargeUpload" }},
    {"testName": {"contains": "CertificateChangedOnDisk"}},
    {"testName": {"contains": "CertificateChangedOnDisk_Symlink"}},
    {"testAssembly": {"contains": "IIS"}}, // IIS is flaky AF
    {"testAssembly": {"contains": "Template"}},
    {"failureMessage": {"contains":"(Site is started but no worker process found)"}},
    {"failureMessage": {"contains": "network disconnected"}}
  ],
  "failOnRules": [
  ],
  "quarantineRules": [
  ]
}
```

File: eng/test-configuration.json [hunk 13]
```diff
@@ -6,20 +6,8 @@
     {"testName": {"contains": "FlakyTestToEnsureRetryWorks" }},
     {"testName": {"contains": "ConfigureEndpointDevelopmentCertificateGetsLoadedWhenPresent" }},
```
Comment (amcasey) [inline eng/test-configuration.json:7]:
```suggestion
    {"testName": {"contains": "ConfigureEndpointDevelopmentCertificateGetsLoadedWhenPresent" }}, // inherently flaky (writes to a well-known path)
```

File: eng/test-configuration.json [hunk 14]
```diff
@@ -4,22 +4,10 @@
   "localRerunCount" : 3,
   "retryOnRules": [
     {"testName": {"contains": "FlakyTestToEnsureRetryWorks" }},
-    {"testName": {"contains": "ConfigureEndpointDevelopmentCertificateGetsLoadedWhenPresent" }},
+    {"testName": {"contains": "ConfigureEndpointDevelopmentCertificateGetsLoadedWhenPresent" }}, // inherently flaky (writes to a well-known path)
```
Comment (amcasey) [inline eng/test-configuration.json:7]:
```suggestion
    {"testName": {"contains": "ConfigureEndpointDevelopmentCertificateGetsLoadedWhenPresent" }}, // writes to a well-known path
```

File: eng/test-configuration.json [hunk 15]
```diff
@@ -4,22 +4,10 @@
...
-    {"testName": {"contains": "TraceIdentifierIsUnique"}},
-    {"testName": {"contains": "LocationChangingHandlers_CannotCancelTheNavigationAsynchronously_UntilReturning"}},
-    {"testName": {"contains": "ClientAbortingConnectionImmediatelyIsNotLoggedHigherThanDebug"}},
-    {"testName": {"contains": "HttpsConnectionClosedWhenResponseDoesNotSatisfyMinimumDataRate"}},
-    {"testName": {"contains": "BidirectionalStream_ServerReadsDataAndCompletes_GracefullyClosed"}},
-    {"testName": {"contains": "POST_ServerAbort_ClientReceivesAbort"}},
-    {"testName": {"contains": "POST_ServerAbortAfterWrite_ClientReceivesAbort"}},
-    {"testName": {"contains": "ServerReset_BeforeRequestBody_ClientBodyThrows"}},
-    {"testName": {"contains": "InjectedStartup_DefaultApplicationNameIsEntryAssembly"}},
     {"testName": {"contains": "CertificateChangedOnDisk"}},
```
Comment (amcasey) [inline eng/test-configuration.json:9]:
```suggestion
    {"testName": {"contains": "CertificateChangedOnDisk"}}, // depends on FS event timing
```

File: eng/test-configuration.json [hunk 16]
```diff
@@ -4,22 +4,10 @@
...
     {"testName": {"contains": "CertificateChangedOnDisk_Symlink"}},
```
Comment (amcasey) [inline eng/test-configuration.json:10]:
```suggestion
    {"testName": {"contains": "CertificateChangedOnDisk_Symlink"}}, // depends on FS event timing
```

File: eng/test-configuration.json [hunk 17]
```diff
@@ -4,22 +4,10 @@
   "localRerunCount" : 3,
   "retryOnRules": [
     {"testName": {"contains": "FlakyTestToEnsureRetryWorks" }},
-    {"testName": {"contains": "ConfigureEndpointDevelopmentCertificateGetsLoadedWhenPresent" }},
+    {"testName": {"contains": "ConfigureEndpointDevelopmentCertificateGetsLoadedWhenPresent" }}, // inherently flaky (writes to a well-known path)
     {"testName": {"contains": "MaxRequestBufferSizeTests.LargeUpload" }},
```
Comment (amcasey) [inline eng/test-configuration.json:8]:
```suggestion
    {"testName": {"contains": "MaxRequestBufferSizeTests.LargeUpload" }}, // tries to manipulate the state of the server's buffer
```

**Review by amcasey — COMMENTED** [review]
Flesh out comments in test-configuration.json, since CI seems happy.

**Comment by amcasey** [comment]
@wtgodbe Any concerns?

**Review by wtgodbe — APPROVED** [review]
Looks great! Thanks for taking the initiative on this.

**Pull request merged**